### DISCUSSION TITLE
Omit password from logs

### DISCUSSION THREAD

**Discussion by cdurl** [thinkst/opencanary#280] [Feature Requests]
Hello, i'm trying to md5 passwords from the logs or at least omit them. I tried to alter the ssh.py auth_password function by casting the password value as an empty string but did not work. Any ideas?

def auth_password(self, packet):
        """
        Password authentication.  Payload::
            string password

        Make a UsernamePassword credential and verify it with our portal.
        """
        password = getNS(packet[1:])[0]
        c = credentials.UsernamePassword(self.user, password)

        us = self.transport.getHost()
        peer = self.transport.getPeer()

        logdata = {'USERNAME': self.user, 'PASSWORD': "password", 'LOCALVERSION': self.transport.ourVersionString, 'REMOTEVERSION': self.transport.otherVersionString}
        logtype =  self.transport.factory.canaryserv

**Comment by jayjb** [comment]
Hi @cdurl,

Its a valid request. I think you are on the right track there though. You could even remove the `getNS(packet[1:])[0]` part entirely and just add some string "PASSWORD" to replace it.
